Compiling error when having Pid Only+Scale selected

Seems to be related to the displaytemplate for scale which has code for water level sensor running but with TOF=0

```
src/Displaytemplatescale.h: In function 'void printScreen()':
Archiving .pio\build\esp32_ota\lib1d8\libTSIC.a
src/Displaytemplatescale.h:146:35: error: 'percentage' was not declared in this scope
             u8g2.printf("%.0f\n", percentage);  // display water level
                                   ^~~~
src/Displaytemplatescale.h:146:35: note: suggested alternative: 'rename'
Compiling .pio\build\esp32_ota\lib67b\Adafruit_VL53L0X\core\src\vl53l0x_api.cpp.o
             u8g2.printf("%.0f\n", percentage);  // display water level
                                   ^~~~
                                   rename
In file included from src/rancilio-pid.cpp:724:
src/brewvoid.h: In function 'void brew()':
src/brewvoid.h:359:34: error: 'firstreading' was not declared in this scope
         if (brewswitch == LOW && firstreading == 0) {
                                  ^~~~
src/brewvoid.h:359:34: note: suggested alternative: 'sigpending'
         if (brewswitch == LOW && firstreading == 0) {
                                  ^~~~
                                  sigpending
*** [.pio\build\esp32_ota\src\rancilio-pid.cpp.o] Error 1
======================================================================================================================================= [FAILED] Took 40.45 seconds =======================================================================================================================================
Environment    Status    Duration

esp32_ota      FAILED    00:00:40.453
================================================================================================================================== 1 failed, 0 succeeded in 00:00:40.453 ======================================================================================
```

userconfig:
```
// Machine

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) MACHINEID 3                // see above list of supported machines

// Display

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) OLED_DISPLAY 2             // 0 = deactivated, 1 = SH1106 (e.g. 1.3 "128x64), 2 = SSD1306 (e.g. 0.96" 128x64)

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) OLED_I2C 0x3C              // I2C address for OLED, 0x3C by default

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) DISPLAYTEMPLATE 4          // 1 = Standard display template, 2 = Minimal template, 3 = only temperature, 4 = scale template, 20 = vertical display (see git Handbook for further information)

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) DISPLAYROTATE U8G2_R0      // rotate display clockwise: U8G2_R0 = no rotation; U8G2_R1 = 90°; U8G2_R2 = 180°; U8G2_R3 = 270°

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) SHOTTIMER 2                // 0 = deactivated, 1 = activated 2 = with scale

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) HEATINGLOGO 0              // 0 = deactivated, 1 = Rancilio, 2 = Gaggia

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) OFFLINEGLOGO 1             // 0 = deactivated, 1 = activated

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) BREWSWITCHDELAY 3000       // time in ms that the brew switch will be delayed (shot timer will show that much longer after switching off)

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) VERBOSE 0                  // 1 = Show verbose output (serial connection), 0 = show less

// Connectivity

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) CONNECTMODE 1              // 0 = offline 1 = WIFI-MODE 2 = AP-MODE (not working in the moment)

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) HOSTNAME "orione"

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) PASS ""        // default password for WiFiManager

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) MAXWIFIRECONNECTS 5        // maximum number of reconnection attempts, use -1 to deactivate

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) WIFICONNECTIONDELAY 10000  // delay between reconnects in ms

// PID & Hardware

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) ONLYPID 1                  // 1 = Only PID, 0 = PID and preinfusion

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) ONLYPIDSCALE 1             // 0 = off , 1 = OnlyPID with Scale

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) BREWMODE 2                 // 1 = Brew by time (with preinfusion); 2 = Brew by weight (from scale)

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) BREWDETECTION 3            // 0 = off, 1 = Software (Onlypid 1), 2 = Hardware (Onlypid 0), 3 = Sensor/Hardware for Only PID

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) BREWSWITCHTYPE 1           // 1 = normal Switch, 2 = Trigger Switch

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) TRIGGERTYPE HIGH           // LOW = low trigger, HIGH = high trigger relay

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) VOLTAGESENSORTYPE LOW     // BREWDETECTION 3 configuration

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) PINMODEVOLTAGESENSOR INPUT // Mode INPUT_PULLUP, INPUT or INPUT_PULLDOWN_16 (Only Pin 16)

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) PRESSURESENSOR 0           // 1 = pressure sensor connected to A0; PINBREWSWITCH must be set to the connected input!

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) TEMPLED 1                  // set led pin high when brew or steam set point is within range

// TOF sensor for water level

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) TOF 0                      // 0 = no TOF sensor connected; 1 = water level by TOF sensor

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) TOF_I2C 0x29               // I2C address of TOF sensor; 0x29 by default

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) CALIBRATION_MODE 0         // 1 = enable to obtain water level calibration values; 0 = disable for normal PID operation; can also be done in Blynk

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) WATER_FULL 102             // value for full water tank (=100%) obtained in calibration procedure (in mm); can also be set in Blynk

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) WATER_EMPTY 205            // value for empty water tank (=0%) obtained in calibration procedure (in mm); can also be set in Blynk

// E-Trigger

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) ETRIGGER 0                 // 0 = no trigger (for Rancilio except Rancilio E), 1 = trigger for CPU of Rancilio E

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) ETRIGGERTIME 600           // seconds, time between trigger signal

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) TRIGGERRELAYTYPE HIGH      // LOW = low trigger, HIGH = high trigger relay for E-Trigger

// Brew Scale

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) SCALE_SAMPLES 2                     // Load cell sample rate

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) SCALE_CALIBRATION_FACTOR 545.00    // Raw data is divided by this value to convert to readable data

/* Pressure sensor
 *

measure and verify "offset" value, should be 10% of ADC bit reading @supply volate (3.3V)
same goes for "fullScale", should be 90% */ [#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) OFFSET 102 // 10% of ADC input @3.3V supply = 102 [#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) FULLSCALE 922 // 90% of ADC input @3.3V supply = 922 [#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) MAXPRESSURE 200
// PlatformIO OTA

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) OTA true                   // true = OTA activated, false = OTA deactivated

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) OTAHOST "orione"           // Name to be shown in Arduino IDE/PlatformIO port

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) OTAPASS ""          // Password for OTA updates

// MQTT

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) MQTT 0                             // 1 = MQTT enabled, 0 = MQTT disabled

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) MQTT_USERNAME "mymqttuser"

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) MQTT_PASSWORD "mymqttpass"

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) MQTT_TOPIC_PREFIX "custom/Küche."  // topic will be "<MQTT_TOPIC_PREFIX><HOSTNAME>/<READING>"

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) MQTT_SERVER_IP "XXX.XXX.XXX.XXX"   // IP-Address of locally installed mqtt server

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) MQTT_SERVER_PORT 1883

// BLYNK

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) BLYNK 0                                 // 0 = no Blynk , 1 = Blynk

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) AUTH ""

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) BLYNKADDRESS "blynk.clevercoffee.de"    // blynk-server IP-Address

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) BLYNKPORT 8080                          // blynk-server portver

// Grafana

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) GRAFANA 1                  // 2 = custom Grafana 1 = Grafana visualisation (access required), 0 = off (default)

// INFLUXDB

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) INFLUXDB 0                 // 1 = INFLUX enabled, 0 = INFLUX disabled

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) INFLUXDB_URL ""            // InfluxDB server address

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) INFLUXDB_AUTH_TYPE 1       // 1 = API Token , 2 = User/Pass

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) INFLUXDB_API_TOKEN ""

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) INFLUXDB_ORG_NAME ""

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) INFLUXDB_USER ""

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) INFLUXDB_PASSWORD ""

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) INFLUXDB_DB_NAME "coffee"  // InfluxDB bucket name

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) INFLUXDB_INTERVAL 5000     // Send interval in milliseconds

// PID Parameters (not yet in Web interface)

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) EMA_FACTOR 0.6             // Smoothing of input that is used for Tv (derivative component of PID). Smaller means less smoothing but also less delay, 0 means no filtering

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) BREWPID_DELAY 0           // delay until enabling PID controller during brew (no heating during this time)

// Backflush values

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) FILLTIME 3000              // time in ms the pump is running

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) FLUSHTIME 6000             // time in ms the 3-way valve is open -> backflush

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) MAXFLUSHCYCLES 5           // number of cycles the backflush should run, 0 = disabled

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) TEMPSENSOR 2               // Temp sensor type: 1 = DS18B20, 2 = TSIC306

// Pin Layout

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) PINTEMPSENSOR 16            // Temp sensor pin

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) PINPRESSURESENSOR 99       // Pressuresensor 0: A0 (ESP8266), >0 ONLY ESP32

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) PINVALVE 97                // Output pin for 3-way-valve

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) PINPUMP 96                 // Output pin for pump

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) PINHEATER 19               // Output pin for heater

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) PINVOLTAGESENSOR 4        // Input pin for voltage sensor (optocoupler to detect brew switch)

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) PINETRIGGER 98             // PIN for E-Trigger relay

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) PINBREWSWITCH 95            // 0: A0 (ESP8266) ; >0 : DIGITAL PIN, ESP32 OR ESP8266: ONLY USE PIN15 AND PIN16!

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) PINSTEAMSWITCH 94          // STEAM active

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) LEDPIN    18               // LED PIN ON near setpoint

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) OLED_SCL 22                 // Output pin for display clock pin

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) OLED_SDA 21                 // Output pin for display data pin

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) HXDATPIN 33                // weight scale PIN

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) HXCLKPIN 32                // weight scale PIN

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) SCREEN_WIDTH 128           // OLED display width, in pixels

[#define](https://chat.rancilio-pid.de/ranciliopid/channels/soforthilfe#) SCREEN_HEIGHT 64           // OLED display height, in pixels
```

--- a/src/Displaytemplatescale.h
+++ b/src/Displaytemplatescale.h
@@ -141,11 +141,11 @@ void printScreen() {
             u8g2.print("Offline Mode");
         }
 
-        if (TOF == 1) {
+        #if TOF == 1
             u8g2.setCursor(100, 2);
             u8g2.printf("%.0f\n", percentage);  // display water level
             u8g2.print((char)37);
-        }
+        #endif
 
         u8g2.sendBuffer();
     }

--- a/src/brewvoid.h
+++ b/src/brewvoid.h
@@ -356,7 +356,7 @@ void brew() {
             weightBrew = weight - weightPreBrew;
         }
 
-        if (brewswitch == LOW && firstreading == 0) {
+        if (brewswitch == LOW && movingAverageInitialized) {
             // check if brewswitch was turned off at least once, last time,
             brewswitchWasOFF = true;
         }

